core/scripts/chaincli/handler/scrape_node_config.go

```diff
@@ -0,0 +1,270 @@
+	log.Warn("This does NOT scrape for payee address, admin address etc. Please verify that manually.")
+	cls := make([]cmd.HTTPClient, len(h.cfg.KeeperURLs))
+	for i := range h.cfg.KeeperURLs {
+		url := h.cfg.KeeperURLs[i]
+		email := h.cfg.KeeperEmails[i]
+		if len(email) == 0 {
+			email = defaultChainlinkNodeLogin
+		}
+		pwd := h.cfg.KeeperPasswords[i]
+		if len(pwd) == 0 {
+			pwd = defaultChainlinkNodePassword
+		}
+
+		cl, err := authenticate(url, email, pwd, log)
+		if err != nil {
+			log.Fatal(err)
+		}
+		cls[i] = cl
+	}
+
+	nodes := map[string]*NodeInfo{}
+	var wg sync.WaitGroup
+	for i, cl := range cls {
+		wg.Add(1)
+		go h.scrapeNodeInfo(&wg, i, cl, nodes, log)
+	}
+	wg.Wait()
+
+	// if node info is not in RDD and weiwatchers, don't proceed further
+	if !h.cfg.VerifyNodes {
+		return
+	}
+	nodeInfos := h.fetchNodeInfos(ctx, log)
+	cnt := 0
+	for _, ni := range nodeInfos {
+		nodeAddr := ni.NodeAddress[0]
+		node := nodes[nodeAddr]
+		if node == nil {
+			continue
+		}
+		cnt++
+
+		log.Infof("start comparing data for node %s", nodeAddr)
+		if node.Equals(ni, log) {
+			log.Infof("node %s info is correct", nodeAddr)
+		} else {
+			log.Errorf("node %s info differs between the node instance and weiwatcher", nodeAddr)
+		}
+	}
+
+	if cnt != len(nodes) {
+		log.Infof("there are %d nodes provisioned , but .env is missing %d nodes", len(nodes), len(nodes)-cnt)
+	}
+}
+
+func (h *baseHandler) fetchNodeInfos(ctx context.Context, log logger.Logger) []NodeInfo {
+	client := http.DefaultClient
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, h.cfg.NodeConfigURL, nil)
+	if err != nil {
+		log.Fatalf("failed to build a GET request: %s", err)
+	}
+
+	resp, err := client.Do(req)
+	if err != nil {
+		log.Fatalf("failed to make a GET request: %s", err)
+	}
+	defer resp.Body.Close()
+
+	var nodeInfos []NodeInfo
+	if err := json.NewDecoder(resp.Body).Decode(&nodeInfos); err != nil {
+		log.Fatalf("failed to read response: %s", err)
+	}
+
+	return nodeInfos
+}
+
+func (h *baseHandler) scrapeNodeInfo(wg *sync.WaitGroup, i int, cl cmd.HTTPClient, nodes map[string]*NodeInfo, log logger.Logger) {
+	defer wg.Done()
+
+	// get node addresses
+	resp, err := nodeRequest(cl, "/v2/keys/eth")
+	if err != nil {
+		log.Fatalf("failed to get ETH keys: %s", err)
+	}
+	var ethKeys cmd.EthKeyPresenters
+	if err = jsonapi.Unmarshal(resp, &ethKeys); err != nil {
+		log.Fatalf("failed to unmarshal response body: %s", err)
+	}
+	var nodeAddresses []string
+	for index := range ethKeys {
+		nodeAddresses = append(nodeAddresses, common.HexToAddress(ethKeys[index].Address).Hex())
+	}
+	ni := &NodeInfo{
+		NodeAddress: nodeAddresses,
+	}
+
+	// get node ocr2 config
+	ocr2Config, err := getNodeOCR2Config(cl)
+	if err != nil {
+		log.Fatalf("failed to get node OCR2 config: %s", err)
+	}
+	ni.Ocr2ID = []string{ocr2Config.ID}
```

Comment:
Can we add a function to the node info type that accepts the ocr2Config as a param, and then performs these operations within the function?

Reply:
Or would it make sense to first perform all three requests, and then update the NodeInfo instance only if all three requests are successful? That way you can avoid partial updates on the NodeInfo, e.g. if the second or third request fails?

Reply:
done